File: tests/PHP/FactoryMethodTest.php
<?php
use PHPUnit\Framework\TestCase;
use Patterns\Creational\FactoryMethod\Factory;

class FactoryMethodTest extends TestCase
{
    public function testFactoryProducesCorrectObject()
    {
        $factory = new Factory();
        $product = $factory->createProduct('ConcreteProduct');
        $this->assertInstanceOf(ConcreteProduct::class, $product);
    }
}

File: tests/Go/factory_method_test.go
package factorymethod

import (
    "testing"
)

func TestFactoryProducesCorrectObject(t *testing.T) {
    factory := Factory{}
    product := factory.CreateProduct("ConcreteProduct")
    if product.Name() != "ConcreteProduct" {
        t.Errorf("Expected ConcreteProduct, got %s", product.Name())
    }
}
